tests/run/test_dataclasses.pyx

```diff
@@ -548,6 +548,48 @@ class C_TestReplace_test_initvar_is_specified:
     def __post_init__(self, y):
         self.x *= y
 
+@dataclass
+@cclass
+class C_TestReplace_test_recursive_repr:
+    f: 'C'
```

Comment:
You're right, but the file is auto-generated (by reprocessing the CPython tests) so it'll need fixing in the script that generates the file rather than here.

Reply:
I'm made all of these typed as `object` (which is enough for the tests to work... I think practically this was how Cython was typing them with the wrong name). Obviously accurate typing would be nicer, but since they're being produced from a fairly dumb conversion script `object` seems universal and good enough